### PR TITLE
Fix ambiguous "Proceed with installation" prompt

### LINKED ISSUE #3429: 'Proceed with installation (default)' prompt is ambiguous
### Problem

After customizing the installation, rustup shows "Proceed with installation (default)" instead of "Proceed with installation (custom)" suggesting it is going to ignore what the user has done and run a default installation instead.

### Steps

1. run rustup
2. when presented with the three options, pick `2) Customize installation`
3. customize the installation
4. observe the label for `1)` after customizing and being back in the menu

### Possible Solution(s)

Change the phrasing to `1) Proceed with installation (custom)` after the user has customized the installation.

### Notes

This is a text bug, not a code bug.

### Rustup version

```console
Windows 10 x64:


>rustup-init --version
rustup-init 1.26.0 (5af9b9484 2023-04-05)
```
```


### Installed toolchains

```console
none, rustup hasn't installed anything yet.
```

### PR DESCRIPTION
**Pull request by brettearle**
Changed test and prompt of default prompt in stdout.
Fix #3429

### CODE REVIEW

**Comment by brettearle** [comment]
Still a WIP, need to refactor a little

**Comment by rami3l** [comment]
@brettearle I'm converting this to a draft since you've declared it to be a WIP. Please feel free to convert it back when you're ready :)

**Comment by brettearle** [comment]
Alright all done @rami3l, unless you need me to change something :)

File: src/cli/self_update.rs [hunk 1]
```diff
@@ -433,10 +433,10 @@ pub(crate) fn install(
         let msg = pre_install_msg(opts.no_modify_path)?;
 
         md(&mut term, msg);
-
+        let mut customised_install: bool = false;
```
Comment (djc) [inline src/cli/self_update.rs]: Please omit the type annotation, which is unnecessary, and rename `customised_install` to `customized_install` to match convention for US English spelling.

File: src/cli/common.rs [hunk 2]
```diff
@@ -56,12 +56,18 @@ pub(crate) enum Confirm {
...
 }
 
-pub(crate) fn confirm_advanced() -> Result<Confirm> {
+pub(crate) fn confirm_advanced(customised_install: &bool) -> Result<Confirm> {
     writeln!(process().stdout().lock())?;
-    writeln!(
-        process().stdout().lock(),
-        "1) Proceed with installation (default - just press enter)"
-    )?;
+    let one_prompt = match customised_install {
```
Comment (djc) [inline src/cli/common.rs]: I'd rather avoid the extra line here.

File: src/cli/common.rs [hunk 3]
```diff
@@ -56,12 +56,18 @@ pub(crate) enum Confirm {
...
+        true => "Proceed with custom install. Press Enter to run custom install",
+        false => "Proceed with standard install. Press Enter to run (1)",
+    };
+    writeln!(process().stdout().lock(), "{one_prompt}")?;
+    let first_option = match customised_install {
+        true => "1) Proceed with customised install",
```
Comment (djc) [inline src/cli/common.rs]: I think we should keep the `(default - just press enter)` and just change the wording before that in the first option. I'd suggest "Proceed with selected options (default - just press enter)" in the customized case.

File: src/cli/self_update.rs (hunk 1 above)
Comment (brettearle) [reply src/cli/self_update.rs]: this has become a enum now. this seems to be outdated

Comment (brettearle) [reply src/cli/self_update.rs]: will change the spelling though

File: src/cli/common.rs (hunk 2 above)
Comment (brettearle) [reply src/cli/common.rs:61]: as in 60?

File: src/cli/common.rs (hunk 3 above)
Comment (brettearle) [reply src/cli/common.rs]: too easy

**Comment by brettearle** [comment]
Ready for review again, happy to change if required :)

File: src/cli/self_update.rs [hunk 4]
```diff
@@ -143,6 +143,11 @@ impl ToString for SelfUpdateMode {
     }
 }
 
+pub enum CustomizedInstall {
```
Comment (djc) [inline src/cli/self_update.rs]: IMO `Yes`/`No` enums like this are an anti-pattern. Instead, it should be `enum Install { Custom, Default }`. But really not sure we need an enum for this case over a `bool`.

Comment (brettearle) [reply src/cli/self_update.rs]: good point. Changed back to bool

Comment (brettearle) [reply src/cli/self_update.rs]: was doing mainly for readability but can see how its bad

File: src/cli/common.rs [hunk 5]
```diff
@@ -12,7 +12,7 @@ use anyhow::{anyhow, Context, Result};
 use git_testament::{git_testament, render_testament};
 use once_cell::sync::Lazy;
 
-use super::self_update::{self, CustomizedInstall};
+use super::self_update::{self};
```
Comment (djc) [inline src/cli/common.rs]: Should remove `::{self}`, this is unnecessary.

File: src/cli/common.rs [hunk 6]
```diff
@@ -56,13 +56,11 @@ pub(crate) enum Confirm {
     Advanced,
 }
 
-pub(crate) fn confirm_advanced(customized_install: &CustomizedInstall) -> Result<Confirm> {
+pub(crate) fn confirm_advanced(customized_install: &bool) -> Result<Confirm> {
```
Comment (djc) [inline src/cli/common.rs]: There's no point in passing `bool` by reference, just pass it directly here.

File: src/cli/common.rs (hunk 5 above)
Comment (brettearle) [reply src/cli/common.rs]: done :)

File: src/cli/common.rs (hunk 6 above)
Comment (brettearle) [reply src/cli/common.rs]: Is that best practice in rust in general? pass reference only when necessary I mean

also done :)

**Review by djc — APPROVED** [review]

File: src/cli/common.rs (hunk 6 above)
Comment (djc) [reply src/cli/common.rs]: Yes -- in particular, `Copy` types (including `bool`, integer and floating point types) are typically passed by-value.

**Comment by djc** [comment]
Please squash these changes into a single commit (I usually do this with `git rebase -i`).

File: src/cli/common.rs (hunk 6 above)
Comment (brettearle) [reply src/cli/common.rs]: Sweet as, going in the anki

**Comment by brettearle** [comment]
Sweet as, will do.

**Comment by brettearle** [comment]
Gday, Is that rebase done the right way? Just unsure about using force to over write it.

**Review by rami3l — APPROVED** [review]
LGTM, thanks a lot!

**Comment by djc** [comment]
Yes, the rebase is great -- thanks!

**Pull request merged**